dask_sql/input_utils/location.py

```diff
@@ -23,20 +31,23 @@ def to_dc(
         gpu: bool = False,
         **kwargs,
     ):
-
         if format == "memory":
             client = default_client()
-            return client.get_dataset(input_item, **kwargs)
+            df = client.get_dataset(input_item, **kwargs)
 
+            for plugin in [DaskInputPlugin(), PandasLikeInputPlugin()]:
+                if plugin.is_correct_input(df, table_name, format, **kwargs):
+                    return plugin.to_dc(df, table_name, format, gpu, **kwargs)
```

Comment:
This seems like the most sensible way to handle published datasets without duplicating code, but I am not sure if there's a cleaner way this could be done.